### PR TITLE
Use entrypoints to register the theme and explicitly state no parallel

### PR DESCRIPTION
**Pull request by Cadair**
This is an attempt to fix the build errors seen on core and ndcube.

The main change is the addition of the `setup(app)` function with the return value.

While I was nobbing around I also added the entry point stuff to automatically register the theme path.

### CODE REVIEW

File: sunpy_sphinx_theme/conf.py [hunk 1]
```diff
@@ -21,24 +23,14 @@ def page_url(page):
...
-    return [str(theme_path)]
-
-
 html_theme = "sunpy"
-html_theme_path = get_html_theme_path()
-templates_path = [os.path.join(html_theme_path[0], html_theme, "templates")]
-html_static_path = [os.path.join(html_theme_path[0], html_theme, "static")]
-html_css_files = [os.path.join(html_static_path[0], "sunpy_style.css")]
-html_extra_path = [os.path.join(html_static_path[0], "img")]
-html_favicon = os.path.join(html_extra_path[0], "favicon-32.ico")
```
Comment (Cadair) [inline sunpy_sphinx_theme/conf.py:38]: I am concerned I am going to have broken something here, local testing on sunpy.org and the ndcube docs indicates it should be ok, but maybe we should do an rc and test it?

**Comment by Cadair** [comment]
pre-commit.ci autofix

File: sunpy_sphinx_theme/conf.py [hunk 2]
```diff
@@ -21,24 +22,18 @@ def page_url(page):
...
-html_theme = "sunpy"
-html_theme_path = get_html_theme_path()
-templates_path = [os.path.join(html_theme_path[0], html_theme, "templates")]
-html_static_path = [os.path.join(html_theme_path[0], html_theme, "static")]
-html_css_files = [os.path.join(html_static_path[0], "sunpy_style.css")]
-html_extra_path = [os.path.join(html_static_path[0], "img")]
-html_favicon = os.path.join(html_extra_path[0], "favicon-32.ico")
+# TODO: Remove these after downstream repos have been fixed to not rely on them
+html_static_path = [str(theme_base_path / "static")]
+html_extra_path = [str(theme_base_path / "static" / "img")]
```
Comment (Cadair) [inline sunpy_sphinx_theme/conf.py:35]: Don't know if you want to keep this hack this or patch the downstream projects. Seems easier to do this even though I don't technically think it's needed and relying on it downstream is probably a little fragile.

**Comment by nabobalis** [comment]
> @nabobalis You wanna take this over the line or you want me to do it?

I'll do it.

**Pull request merged**